src/commands/getDefaultWebAppToDeploy.ts

```diff
@@ -27,7 +20,8 @@ export async function getDefaultWebAppToDeploy(context: IDeployWizardContext): P
             return <WebAppTreeItem>defaultWebApp;
         } else {
             // if defaultPath or defaultNode cannot be found or there was a mismatch, delete old settings and prompt to save next deployment
-            workspaceConfig.update(configurationSettings.defaultWebAppToDeploy, undefined, context.configurationTarget);
+            context.fsPath ? await updateWorkspaceSetting(configurationSettings.defaultWebAppToDeploy, undefined, context.fsPath) :
```

Comment:
Yeah double check `context.configurationTarget` for this file, too. I don't understand how you were refactoring, but I don't think it's correct